juju_verify/verifiers/ovn_central.py

```diff
@@ -0,0 +1,399 @@
+from juju_verify.verifiers.result import Result, Severity, checks_executor
+
+logger = logging.getLogger(__name__)
+
+StatusCollectionType = Dict[str, "UnitClusterStatus"]
+
+
+class ClusterStatus:  # pylint: disable=R0902
+    """Object representation of OVN cluster status.
+
+    Attributes of this object represent status of a single (Southbound or Northbound)
+    cluster as retrieved from 'cluster-status' action of ovn-central charm unit. Example
+    of an input data for initialization of this class:
+
+    .. code-block:: yaml
+
+        Cluster ID: 567e7225-369e-40d6-abf8-9b442bbcd18b
+        Server ID: 16335def-c21e-404c-b123-8337b3013c07
+        Address: ssl:10.5.2.232:6644
+        Status: cluster member
+        Role: follower
+        Term: 34
+        Leader: dbdb
+        Vote: dbdb
+        Log: '[66, 66]'
+        Entries not yet committed: 0
+        Entries not yet applied: 0
+        Servers:
+          dbdb:
+            Address: ssl:10.5.0.144:6644
+            Unit: ovn-central/7
+          f1a2:
+            Address: ssl:10.5.3.200:6644
+            Unit: ovn-central/10
+          '1633':
+            Address: ssl:10.5.2.232:6644
+            Unit: ovn-central/6
+    """
+
+    def __init__(self, raw_status: str) -> None:
+        """Initialize Cluster status class.
+
+        For more info about the purpose and input data format see class documentation.
+
+        :param raw_status: YAML string containing status of a single OVN cluster.
+        """
+        try:
+            status_dict = yaml.safe_load(raw_status)
+            if not isinstance(status_dict, dict):
+                raise ValueError("Input data is not a YAML dictionary.")
+        except (yaml.YAMLError, ValueError) as exc:
+            raise JujuVerifyError(
+                f"Failed to load OVN status as YAML:\n{raw_status}"
+            ) from exc
+
+        try:
+            self.cluster_id = status_dict["Cluster ID"]
+            self.server_id = status_dict["Server ID"]
+            self.status = status_dict["Status"]
+            self.role = status_dict["Role"]
+            self.term = status_dict["Term"]
+            self.leader = status_dict["Leader"]
+            self.vote = status_dict["Vote"]
+            self.log = status_dict["Log"]
+            self.entries_not_committed = status_dict["Entries not yet committed"]
+            self.entries_not_applied = status_dict["Entries not yet applied"]
+            self.servers = status_dict["Servers"]
+
+        except KeyError as exc:
+            raise JujuVerifyError(
+                f"Failed to deserialize OVN cluster status. Missing "
+                f"'{exc.args[0]}' from cluster status data."
+            ) from exc
+
+    @property
+    def short_id(self) -> str:
+        """Return first 4 characters of a Server ID.
+
+        This format is often used to refer to servers within cluster instead of full ID.
+        """
+        return self.server_id[:4]
```

Comment:
We never verified that `Server ID` is string, what if it will be an integer?

Reply:
`Server ID` is UUID4 string (with dashes) like this `1c6aff37-45c2-4d4e-aeab-a057cb240cbb`.

Reply:
Ok, thanks for clarification.